src/ipred_prepare.rs

```diff
@@ -183,9 +186,10 @@ pub fn rav1d_prepare_intra_edges<BD: BitDepth>(
     {
         let px_have = cmp::min(8 * tw, 4 * (w - x)) as usize;
         let n = px_have + have_left as usize;
-        if let Some(prefilter_toplevel_sb_edge) = prefilter_toplevel_sb_edge {
-            let offset = (x * 4) as usize - have_left as usize;
-            &prefilter_toplevel_sb_edge[offset..][..n]
+        if let Some((edge_buf, base)) = prefilter_toplevel_sb_edge {
+            let offset = ((x * 4) as usize - have_left as usize).wrapping_add_signed(base);
+            edge_buf_guard = edge_buf.slice_as(offset..offset + n);
```

Comment:
After #988, I think we can do this:

```suggestion
            edge_buf_guard = edge_buf.slice_as((offset.., ..n));
```

Reply:
Oh, okay.  Damnit.  But yeah it's fine as is then.